src/NServiceBus.FileBasedRouting/XmlRoutingFileAccess.cs

```diff
@@ -1,27 +1,23 @@
 ﻿namespace NServiceBus.FileBasedRouting
 {
     using System;
-    using System.IO;
     using System.Xml;
     using System.Xml.Linq;
 
     class XmlRoutingFileAccess
     {
-        public string FilePath { get; }
+        public Uri FileUri { get; }
 
-        public XmlRoutingFileAccess(string filePath)
+        public XmlRoutingFileAccess(Uri fileUri)
         {
-            FilePath = filePath;
+            FileUri = fileUri;
         }
 
         public XDocument Read()
         {
             try
             {
-                using (var fileStream = File.OpenRead(FilePath))
-                {
-                    return XDocument.Load(fileStream);
-                }
+                return XDocument.Load(FileUri.ToString());
```

Comment:
> Why do we need to use the Uri type if, in the end, we call ToString() on it?

strictly speaking no, since XDocument does use a custom Uri resolver to avoid certain security risks. It does convert it to an Uri internally though and I wanted to avoid a public api which takes either a path or an uri, as we support relative paths. Uri provides some built-in verification of valid Uris, so you would immediately get exceptions when trying to use invalid Uris during configuration time.

> can this become a security problem (loading XML from a URL)?

Uris aren't tied to http/https, so I don't think we should enforce any protocol. Afaik XDocument has some built in security related logic about loading documents from an Uri, but tbh not sure how far that goes and what doors this would open. Do we have someone more experienced with that to ping?